src/services/users/controllers/deleteAccount.ts

```diff
@@ -4,21 +4,123 @@ import { ObjectId } from 'mongodb';
+      .service(service.structures)
+      .Model.accessibleBy(req.ability, action.read)
+      .countDocuments({
+        'contact.email': user.name,
+      });
+    if (structure > 0) {
+      return true;
+    }
+    return false;
+  };
 
 const deleteAccount =
   (app: Application) => async (req: IRequest, res: Response) => {
     const idUser = req.params.id;
+    const { actionSuppression } = req.query;
+    let user: undefined | IUser;
     try {
-      // xxx pour le moment on n'autorise que les admins à supprimer un user
-      // à voir : suppression d'un rôle et non du user complet si multi rôle
-      await app
-        .service(service.users)
-        .Model.accessibleBy(req.ability, action.delete)
-        .deleteOne({
-          _id: new ObjectId(idUser),
-          roles: { $all: ['admin'] },
-        });
-      res.status(200).json({ deleteSuccess: true, idUser });
+      if (actionSuppression === 'tous') {
+        const isContactStructure = await checkIfUserIsContactStructure(
+          app,
+          req,
+        )(idUser);
+        if (isContactStructure) {
+          return res.status(409).json({
+            message: 'le gestionnaire ne peut pas être supprimé',
+          });
+        }
+        await app
+          .service(service.users)
+          .Model.accessibleBy(req.ability, action.delete)
+          .deleteOne({ _id: new ObjectId(idUser) });
+      } else {
+        let query = {};
+        switch (actionSuppression) {
+          case 'grandReseau':
+            query = {
+              $unset: {
+                reseau: '',
+              },
+              $pull: {
+                roles: 'grandReseau',
+              },
+            };
+            break;
+          case 'structure' || 'structure_coop':
+            // eslint-disable-next-line no-case-declarations
+            const isContactStructure = await checkIfUserIsContactStructure(
+              app,
+              req,
+            )(idUser);
+            if (isContactStructure) {
+              return res.status(409).json({
+                message:
+                  'le gestionnaire ne peut pas perdre son rôle structure',
+              });
+            }
+            query = {
+              $unset: {
+                entity: '',
+              },
+              $pull: {
+                roles: { $in: ['structure', 'structure_coop'] },
+              },
+            };
+            break;
+          case 'prefet':
+            query = {
+              $unset: {
+                departement: '',
+                region: '',
+              },
+              $pull: {
+                roles: 'prefet',
+              },
+            };
+            break;
+          case 'hub_coop':
+            query = {
+              $unset: {
+                hub: '',
+              },
+              $pull: {
+                roles: 'hub_coop',
+              },
+            };
+            break;
+          case 'admin' || 'admin_coop':
```

Comment:
idem invalide mais pas utile le 'admin_coop'
```suggestion
          case 'admin':
```